src/migrations/1.2104-000-settings_parent_type.cr

```diff
@@ -0,0 +1,25 @@
+require "../migration"
+
+module Migrations::SettingParentType
+  include Migration::Irreversible
+
+  def self.up
+    raw_query do |r|
+      r
+        .table(PlaceOS::Model::Settings.table_name)
+        .filter { |s| s["parent_type"].type_of.eq("NUMBER") }
+        .update { |s|
+          {% begin %}
+            {% type = PlaceOS::Model::Settings::ParentType %}
+            r.branch(
+              {% for t in type.constants %}
+                s["parent_type"].eq({{type}}::{{t}}.to_i),
+                {:parent_type => JSON.parse({{type}}::{{t}}.to_json)},
```

Comment:
Does this require a JSON::Any here?
```suggestion
                {:parent_type => {{type}}::{{t}}.to_json},
```

Reply:
Not a JSON any, just something that will encode properly. I previously had as you're suggesting: 9c1c5e802846ccbd04aa7708e14bd38c389eb811. This was resulting in a extra wrap of encoding (included quotes in the string). 

This feel _extremely_ dirty to me so very open to suggestions on an alternative approach, that doesn't involve reimplementing / assuming what that JSON serialising looks like.

Reply:
There totally should be a method in enum that gives the serialised value.
Currently, `to_s` is just the enum's member, while `to_json` is the underscored value.

Preferably...
```suggestion
                {:parent_type => {{type}}::{{t}}.to_s.underscore},
```

Reply:
That's what I was hoping to avoid as it doubles up the implementation. Piping it through `#to_json` ensures it follows the externally defined serialisation.